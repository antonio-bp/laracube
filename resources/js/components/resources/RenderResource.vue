<template>
    <div class="fill-height">
        <div v-if="resource.component === 'table'" class="fill-height">
            <laracube-table :resource="resource"></laracube-table>
        </div>
        <div v-if="resource.component === 'card'" class="fill-height">
            <card :resource="resource"></card>
        </div>
    </div>
</template>

<script>
import LaracubeTable from '@/components/resources/Table';
import Card from '@/components/resources/Card';

export default {
    name: 'RenderResource',
    components: { Card, LaracubeTable },
    props: {
        resource: { required: true },
    },
};
</script>
