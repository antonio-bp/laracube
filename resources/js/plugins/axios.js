import ax from 'axios';
import store from '@/store';

ax.interceptors.request.use(
    function (config) {
        let source = ax.CancelToken.source();
        config.cancelToken = source.token;
        store.commit('axios/ADD_CANCEL_TOKEN', source);
        return config;
    },
    function (error) {
        return Promise.reject(error);
    },
);

export const axios = ax;

export default {
    install(Vue, options) {
        Vue.prototype.$axios = ax;
    },
};
