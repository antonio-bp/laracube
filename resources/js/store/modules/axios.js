export default {
    namespaced: true,
    state: {
        cancelTokens: [],
    },
    getters: {
        cancelTokens(state) {
            return state.cancelTokens;
        },
    },
    mutations: {
        ADD_CANCEL_TOKEN(state, token) {
            state.cancelTokens.push(token);
        },
        CLEAR_CANCEL_TOKENS(state) {
            state.cancelTokens = [];
        },
    },
    actions: {
        CANCEL_PENDING_REQUESTS(context) {
            context.state.cancelTokens.forEach((request, i) => {
                if (request.cancel) {
                    request.cancel();
                }
            });
            context.commit('CLEAR_CANCEL_TOKENS');
        },
    },
};
