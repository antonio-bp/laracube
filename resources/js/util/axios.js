import store from '@/store';

window.axios = require('axios');

window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';

window.axios.interceptors.request.use(
    function (config) {
        let source = window.axios.CancelToken.source();
        config.cancelToken = source.token;
        store.commit('axios/ADD_CANCEL_TOKEN', source);
        return config;
    },
    function (error) {
        return Promise.reject(error);
    },
);
