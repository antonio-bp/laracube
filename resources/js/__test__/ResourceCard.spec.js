import { createLocalVue, mount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';
import RenderResource from '@/components/resources/RenderResource';
import flushPromises from 'flush-promises';
import axios from 'axios';

Vue.use(Vuetify);
const localVue = createLocalVue();

jest.mock('axios', () => ({
    get: () =>
        Promise.resolve({
            data: {
                line1: {
                    value: 100,
                },
                line2: {
                    value: 'from 50',
                },
                trend: {
                    value: 20,
                    icon: 'fa-arrow-up',
                },
                sparkline: {
                    value: [1, 2, 3, 4, 5, 6],
                },
            },
        }),
}));

function factory() {
    let vuetify = new Vuetify();

    return mount(RenderResource, {
        localVue,
        vuetify,
        propsData: {
            resource: {
                columns: 4,
                component: 'big-number',
                heading: 'Heading',
                subHeading: 'Sub Heading',
                type: null,
                uriKey: 'big-number',
            },
        },
        mocks: {
            $axios: axios,
        },
    });
}

describe('BigNumber', () => {
    it('should load the components', () => {
        const wrapper = factory();
        expect(wrapper.find('.v-card').vm).toBeTruthy();
    });

    it('should show the heading and sub-heading', () => {
        const wrapper = factory();
        expect(wrapper.find('.v-card__title').html()).toContain('Heading');
        expect(wrapper.find('.v-card__subtitle').html()).toContain('Sub Heading');
    });

    it('should show the correct data from api response for line 1', async () => {
        const wrapper = factory();
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.find('.line1').html()).toContain(100);
    });

    it('should show the correct data from api response for line 2', async () => {
        const wrapper = factory();
        await flushPromises();
        expect(wrapper.find('.line2').html()).toContain('from 50');
    });

    it('should show the correct data from api response for trend', async () => {
        const wrapper = factory();
        await flushPromises();
        expect(wrapper.find('.trend').html()).toContain('20');
    });
});
