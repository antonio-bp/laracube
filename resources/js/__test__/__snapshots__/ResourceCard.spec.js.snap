// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigNumber should show the correct data from api response for line 1 1`] = `
<div class="fill-height">
  <!---->
  <!---->
</div>
`;
